### PR TITLE
Added largest blob crop transform

### PR DESCRIPTION
**Pull request by antdood**
## What changes were proposed in this pull request?

Made a transform function for cropping the largest blob from an image.

## How was this patch tested?

I ran it with MNIST and Cifar-10 datasets and used ShowImageTransform to validify the crops. Also used drawContours() from the OpenCV API to confirm the right blob was being cropped.

### CODE REVIEW

File: datavec-data/datavec-data-image/src/main/java/org/datavec/image/transform/LargestBlobCropTransform.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import org.datavec.image.data.ImageWritable;
+import org.datavec.image.transform.BaseImageTransform;
+import org.jfree.util.Log;
+import org.nd4j.linalg.factory.Nd4j;
+import java.util.Random;
+import static org.bytedeco.javacpp.opencv_core.*;
+import static org.bytedeco.javacpp.opencv_imgproc.*;
+
+public class LargestBlobCropTransform extends BaseImageTransform<Mat> {
+
```
Comment (agibsonccc) [inline datavec-data/datavec-data-image/src/main/java/org/datavec/image/transform/LargestBlobCropTransform.java]: Javadoc explanations would be really helpful here.

File: datavec-data/datavec-data-image/src/main/java/org/datavec/image/transform/LargestBlobCropTransform.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    protected org.nd4j.linalg.api.rng.Random rng;
+
+    protected int mode,method,blurWidth,blurHeight,upperThresh,lowerThresh,cannyOrThresh;
+
```
Comment (agibsonccc) [inline datavec-data/datavec-data-image/src/main/java/org/datavec/image/transform/LargestBlobCropTransform.java]: May not want to abbreviate variable names here (clarity over brevity in this case)

File: datavec-data/datavec-data-image/src/main/java/org/datavec/image/transform/LargestBlobCropTransform.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    public LargestBlobCropTransform() {
+        this(null);
+    }
+
+    public LargestBlobCropTransform(Random random) {
+        this(random, CV_RETR_CCOMP, CV_CHAIN_APPROX_SIMPLE, 3, 3, 100, 200 , 0);
+    }
+
+    public LargestBlobCropTransform(Random random, int mode, int method, int blurWidth, int blurHeight, int UpperThresh, int LowerThresh, int CannyOrThresh){
+        super(random);
```
Comment (agibsonccc) [inline datavec-data/datavec-data-image/src/main/java/org/datavec/image/transform/LargestBlobCropTransform.java]: Wondering if a lombok style builder would be better here (Just add @Builder to the  annotation. This is a lot of variable arguments.

**Comment by agibsonccc** [comment]
Needs a unit test or 2 as well.

**Comment by agibsonccc** [comment]
Any update here? We're almost ready to merge this.

**Comment by huitseeker** [comment]
[Multimedia support for the Transform process](https://trello.com/c/tyLqHQPH/4-multimedia-support-for-the-transform-process)

**Comment by huitseeker** [comment]
Closed in favor of #313

**Pull request closed without merging**